src/adagrams.js

```diff
@@ -1,8 +1,170 @@
+  } 
+  else {
+    return false;
+  }
+}
+return true;
+},
+
+scoreWord(word){ //The  // Wave 3
+  const valueOne = [ "A", "E", "I", "O", "U", "L", "N", "R", "S", "T"	]; //Arrays, things happen inside the function     
+  const valueTwo = ["D", "G"];	     
+  const valueThree = ["B", "C", "M", "P"];	     
+  const valueFour = ["F", "H", "V", "W", "Y"];	     
+  const valueFive = ["K"];	     
+  const valueEight = ["J", "X"];	     
+  const valueTen = ["Q", "Z"];	          
+  let splitWord = word.toUpperCase().split("");   // ["T", "H", "E"] 
+  let wordLength = word.length;
+  let score = 0;     
+  
+  for (let letter of splitWord) {//E  -  score = 6
+    if(valueOne.indexOf(letter)>-1){
+      score = score + 1;
+    }
+    if(valueTwo.indexOf(letter)>-1){
+      score = score + 2;
+    }
+    if(valueThree.indexOf(letter)>-1){
+      score = score + 3;
+    }
+    if(valueFour.indexOf(letter)>-1){
+      score = score + 4;
+    }
+    if(valueFive.indexOf(letter)>-1){
+      score = score + 5;
+    }
+    if(valueEight.indexOf(letter)>-1){
+      score = score + 8;
+    }
+    if(valueTen.indexOf(letter)>-1){
+      score = score + 10;
+    }
+  }
+
+  if(wordLength >= 7 && wordLength <= 10) {
+    score = score + 8;
+  }
+
+  return score;
   },
-};
 
-// Do not remove this line or your tests will break!
-export default Adagrams;
+  highestScoreFrom(words) { //gets an array of strings 
+      let maxScore = 0;
+      let maxRep = 0;
+      let wordMaxScore = "";
+      let list = [];
+
+      for(let word of words) {//build a "hash" with word, score and length 
+        list.push({
+          word:word.toUpperCase(),
+          score: this.scoreWord(word.toUpperCase()),
+          length:word.length
+        });
+      }
+      for(let item of list) { //get max score
+        if(item.score>maxScore){
+          maxScore = item.score;
+        }
+      }
+      for(let item of list) { //how many times does max score repeats
+        if(item.score == maxScore){
+          maxRep = maxRep + 1;//maxRep++;
+          wordMaxScore = item.word;
+        }
+      }
+      if(maxRep === 1) {// if one, we return winner 
+        return {word:wordMaxScore, score:maxScore};
+      }
+      else{ 
+        let wordArray = [];
+          for(let item of list){ //lets get those max
+            if(item.score==maxScore){
+              wordArray.push(item);
+            }
+          }
+          wordArray.sort((a, b) => (a.length > b.length) ? 1 : -1); //asc de min to max
+          let minLength = wordArray[0].length; //lenght min
+          let maxLength = wordArray[wordArray.length-1].length; //length max
+          let maxRepMinLength = 0;
+          let maxRepMaxLenght = 0;
+          
+          for(let item of wordArray) {// repeats ?
+            if(item.length == minLength){
+              maxRepMinLength = maxRepMinLength + 1;//maxRep++;
+              wordMaxScore = item.word;
```

Comment:
This piece here is kind of verbose. I think once you've found the shortest, you return that, or you return the one whose length == 10